app/models/in_person_enrollment.rb

```diff
@@ -12,6 +12,34 @@ class InPersonEnrollment < ApplicationRecord
 
   validate :profile_belongs_to_user
 
+  # Find enrollments that need a status check via the USPS API
+  def self.needs_usps_status_check check_interval
+    where(status: :pending).
+    and(
+      where(status_check_attempted_at: check_interval).
+      or(where(status_check_attempted_at: nil)),
+    ).
+    order(status_check_attempted_at: :asc)
+  end
+
+  # Does this enrollment need a status check via the USPS API?
+  def needs_usps_status_check? check_interval
+    status == :pending && (
+      status_check_attempted_at.nil? ||
+      check_interval.cover?(status_check_attempted_at)
+    )
+  end
+
+  # Returns the value to use for the USPS enrollment ID
+  def usps_unique_id
+    user_id.to_s
```

Comment:
I believe that we tend to avoid sharing the internal numeric user ID with any systems outside Login.gov (citation needed). Could we consider using UUID, or add a new field to the enrollment?

If I recall correctly, I think USPS's limitation may prevent us from using UUID, so I do wonder about a new field with a uniqueness constraint. Had we considered that before?

Reply:
I think adding a new field to the enrollment is the right way to go. I'm working on a new ticket for this, that I'll pick up immediately.

Reply:
+1 yeah we try not to share the integer ID at all (hence why we log using the string UUID). Let's definitely find another identifier, and it would be great if it was more random/nonsequential (so we can be more certain we don't cross staging/prod IDs, or query the wrong table, etc)